src/events/supportInteractionCreate.event.ts

```diff
@@ -0,0 +1,87 @@
+import {
+    Interaction,
+    MessageActionRow,
+    Modal,
+    ModalActionRowComponent,
+    TextChannel,
+    TextInputComponent,
+} from "discord.js";
+import { Event } from "../handlers/EventHandler";
+
+export default class InteractionCreate extends Event<"interactionCreate"> {
+    public event = "interactionCreate";
+
+    public async run(interaction: Interaction) {
+        if (interaction.isButton()) {
+            const buttonId = interaction.customId;
+            if (buttonId == "get-support") {
+                const modal = new Modal();
+                modal.setCustomId("support-modal");
+                modal.setTitle("New support request");
+
+                const titleInput = new TextInputComponent();
+                titleInput.setCustomId("support-title");
+                titleInput.setLabel("Issue");
+                titleInput.setStyle("SHORT");
+
+                const descriptionInput = new TextInputComponent();
+                descriptionInput.setCustomId("support-description");
+                descriptionInput.setLabel("Description");
+                descriptionInput.setStyle("PARAGRAPH");
```

Comment:
Just wondering, does this allow them to drop images in? Or is it a text only area?

Reply:
Discord its modal feature does not support adding anything else than text fields yet. Maybe in the future